### DISCUSSION TITLE
Is it possible to use categorical features on Apple Silicon?

### DISCUSSION THREAD

**Discussion by gottogethelp** [catboost/catboost#2661] [General]
On the [categorical features page](https://catboost.ai/en/docs/features/categorical-features) there is the following section:

> By default, CatBoost uses one-hot encoding for categorical features with a small amount of different values in most modes. It is not available if training is performed on CPU in
> 
> Pairwise scoring
> 
> The following loss functions use Pairwise scoring:
> 
> YetiRankPairwise
> PairLogitPairwise
> QueryCrossEntropy
>
> Pairwise scoring is slightly different from regular training on pairs, since pairs are generated only internally during the training for the corresponding metrics. One-hot encoding is not available for these loss functions.

However, further down it says:

> ...all categorical features are ignored if training is performed on CPU

I read the first section as categorical features can be used in some circumstances when training on a CPU but the second section seems to say they can never be used when training on a CPU? Which of these is correct?

If the second section is correct then should I assume there is no way to use categorical features on Apple Silicon because [the GPU isn't yet supported](https://github.com/catboost/catboost/issues/2075)?

**Comment by ek-ak** [comment]
Hello!
Quote 
> ...all categorical features are ignored if training is performed on 
is taken from the paragraph about training on dataset **without  target data (labels)** (when only pairs are available) - and is only applicable for pairwise scoring and datasets without  labels.

If it is not your case - you can use categorical features on Apple Silicon.
